contracts/vaults/Claim.sol

```diff
@@ -33,43 +33,70 @@ contract Claim is Base {
         string calldata _descriptionHash)
     external
     onlyCommittee(_pid)
-    noSubmittedClaims(_pid) {
+    noSubmittedClaims(_pid)
+    {
         require(_beneficiary != address(0), "HVE04");
         // require we are in safetyPeriod
         // solhint-disable-next-line not-rely-on-time
         require(block.timestamp % (generalParameters.withdrawPeriod + generalParameters.safetyPeriod) >=
         generalParameters.withdrawPeriod, "HVE05");
         require(_bountyPercentage <= bountyInfos[_pid].maxBounty, "HVE06");
-
-        submittedClaims[_pid] = SubmittedClaim({
+        uint256 claimId;
+        uint256 nonce;
+        do
+        {
+            claimId = uint256(keccak256(abi.encodePacked(_pid, block.number, nonce)));
+            nonce++;
+        } while (claimId == 0 || submittedClaims[claimId].beneficiary != address(0));
+        submittedClaims[claimId] = SubmittedClaim({
+            pid: _pid,
             beneficiary: _beneficiary,
             bountyPercentage: _bountyPercentage,
             committee: msg.sender,
             // solhint-disable-next-line not-rely-on-time
-            createdAt: block.timestamp
+            createdAt: block.timestamp,
+            isChallenged: false
         });
-        emit SubmitClaim(_pid, msg.sender, _beneficiary, _bountyPercentage, _descriptionHash);
+        activeClaims[_pid] = claimId;
+        emit SubmitClaim(
+            _pid,
+            claimId,
+            msg.sender,
+            _beneficiary,
+            _bountyPercentage,
+            _descriptionHash
+        );
+    }
+
+    function challenge(uint256 _claimId) onlyArbitrator external {
```

Comment:
still think it should be called challengeClaim, for clarity and consistency

Reply:
yes

Reply:
thanks :)